bundles/org.openhab.binding.zwavejs/src/main/java/org/openhab/binding/zwavejs/internal/handler/ZwaveJSNodeHandler.java

```diff
@@ -0,0 +1,667 @@
+
+    private @Nullable Object handlePercentTypeCommand(ZwaveJSChannelConfiguration channelConfig, Channel channel,
+            @Nullable ColorCapability colorCap, boolean isColorChannelCommand, PercentType percentTypeCommand) {
+        if (isColorChannelCommand && colorCap != null) {
+            HSBType hsb = new HSBType(colorCap.cachedColor.getHue(), colorCap.cachedColor.getSaturation(),
+                    percentTypeCommand);
+            return handleHSBTypeCommand(channelConfig, channel, colorCap, true, hsb);
+        }
+
+        // For non-color channels, handle inversion and dimmer 100% edge case
+        int value = percentTypeCommand.intValue();
+        if (channelConfig.inverted) {
+            value = 100 - value;
+        }
+        // For dimmers, 100% is often represented as 99
+        if (CoreItemFactory.DIMMER.equals(channelConfig.itemType) && value == 100) {
+            value = 99;
+        }
+        return value;
+    }
+
+    private @Nullable Object handleHSBTypeCommand(ZwaveJSChannelConfiguration channelConfig, Channel channel,
+            @Nullable ColorCapability colorCap, boolean isColorChannelCommand, HSBType hsbTypeCommand) {
+        Object retValue = null;
+        if (isColorChannelCommand && colorCap != null) {
+            colorCap.cachedColor = hsbTypeCommand;
+        }
+        HSBType hsbFull = new HSBType(hsbTypeCommand.getHue(), hsbTypeCommand.getSaturation(), PercentType.HUNDRED);
+        int[] rgb = ColorUtil.hsbToRgb(hsbFull);
+
+        if (channel.getUID().getId().contains(HEX)) {
+            retValue = "%02X%02X%02X".formatted(rgb[0], rgb[1], rgb[2]);
+        } else {
+            Map<String, Integer> colorMap = new HashMap<>();
+            colorMap.put(RED, rgb[0]);
+            colorMap.put(GREEN, rgb[1]);
+            colorMap.put(BLUE, rgb[2]);
+            if (colorCap != null && colorCap.supportsColdWhite) {
+                colorMap.put(COLD_WHITE, 0);
+            }
+            if (colorCap != null && colorCap.supportsWarmWhite) {
+                colorMap.put(WARM_WHITE, 0);
+            }
+            retValue = colorMap;
+        }
+        if (isColorChannelCommand) {
+            // schedule brightness command(s) for dimmer channel(s)
+            PercentType brightness = hsbTypeCommand.getBrightness();
+            thing.getChannels().stream().filter(c -> CoreItemFactory.DIMMER.equals(c.getAcceptedItemType()))
+                    .forEach(c -> {
+                        if (c.getConfiguration()
+                                .as(ZwaveJSChannelConfiguration.class) instanceof ZwaveJSChannelConfiguration cf
+                                && cf.endpoint == channelConfig.endpoint) {
+                            scheduler.submit(() -> handleCommand(c.getUID(), brightness));
+                        }
+                    });
+        }
+
+        return retValue;
+    }
+
+    private @Nullable Object handleQuantityTypeCommand(ZwaveJSChannelConfiguration channelConfig,
```

Comment:
I think this has a bug. It doesn't return a quantity if the factor is 1.0. And probably you need to do the scaling before creating the quantity. (Think fahrenheit..)